packages/extension-base/src/background/handlers/Extension.ts

```diff
@@ -408,13 +414,14 @@ export default class Extension {
         .createType('ExtrinsicPayload', payload, { version: payload.version })
         .sign(pair);
 
-    if (savePass) {
+    if (savePass && password) {
       // unlike queued.account.address the following
       // address is encoded with the default prefix
       // which what is used for password caching mapping
-      this.#cachedUnlocks[pair.address] = Date.now() + PASSWORD_EXPIRY_MS;
+
+      await chromeStorage.setItem('password-cache', (savedPasswords) => ({ ...savedPasswords, [pair.address]: { password, expiresAt: Date.now() + PASSWORD_EXPIRY_MS } }));
```

Comment:
could you break lines longer than 140 (or if you don't mind we have e.g. 120 char limit in common)? My laptop screen adds a scroll on longer lines and I think it also affects the readability :sweat_smile: